docs/P1.ipynb

```diff
@@ -0,0 +1,593 @@
tN3xG83Msvv6w1a9bo4YcfVkxMjPr06eM+8+ByaT/uhgwZojp16mSYX7Ro0SuuBwCygj4KcPOijwJ/QPgE/I2aNGmiadOmacmSJSpevHiGowDTpk1T69atNWzYMPe09evXS7o0mODltzu+++679csvv3jsjPbv36+33npLAwcOVL58+QxuDfDPcNddd8lut+v3339XvXr13NM//PBD/fLLL1f84XU1Z86c0YIFCzR27Fg9+OCD7um//vqrQkJCMixfsGBBFShQQIcPH/Zo76tWrdK6des0atSo664DAKRHHwXIeeijICdhwHHgb1SuXDndeeedevvttzOczi5dOjKwc+dO/fTTT/rtt980a9YszZ49W9Jfp8Gm17VrV+3atUvDhg3T/v379d133+mZZ57RwYMHVbJkSdObA/wj5MuXT+3atdP48eP18ccf6/Dhw1q8eLHeeustFS5cOFtl5s2bV/ny5dP69evdl6689NJL+umnnzJt6zabTU8++aQ++ugjzZ49W7/99pvWrVunV199Vblz5/Y4sgkA2UEfBch56KMgJ+HMJ+Bv1qRJE02ZMsXjSEKal156SS+//LI6dOig4OBglS1bVqNHj9aAAQP0n//8J8N11pGRkZo+fbrGjx+vli1bKiQkRLVr19Zzzz3HFz3gRUOGDFFYWJjGjx+vEydOqGjRourXr5+6d++erfKCgoI0fvx4jRw5Us2aNdMtt9yimjVrauDAgZo6daqSkpIyvKZr167KlSuXPvroI40cOVK33Xab2rRpo379+t3o5gGAJPooQE5EHwU5hc3iXqcAAAAAAAAwhMvuAAAAAAAAYAzhEwAAAAAAAIwhfAIAAAAAAIAxhE8AAAAAAAAwhvAJAAAAAAAAxhA+AQAAAAAAwBjCJwAAAAAAABhD+AQAAAAAAABjCJ8A4Br279+v4cOHq1GjRqpcubKqVaumdu3aae7cuXI6nX9bPSIiIjRx4sS/bX0AAODmRh8FQE5h93UFAOBmtmrVKg0ZMkSlS5dWly5dVKpUKV24cEFffvml3njjDX399deaPHmybDabr6sKAAD+QeijAMhJCJ8A4Ar279+vIUOGKDo6WuPGjZPd/tdX5gMPPKCaNWuqX79+Wr16tR588EEf1hQAAPyT0EcBkNNw2R0AXMH06dMVEBCgYcOGeXTq0jRq1EgPP/yw+7nL5dK0adPUoEEDVahQQY0aNdJHH33k8ZqOHTvqxRdf1LRp01S3bl1VrFhR7dq1048//uix3NatW9W2bVtVrlxZjRo10rfffpth/cnJyRo9erQeeOABVahQQc2aNdOqVas8lomJidEbb7yhJ554QpUqVdKLL754A+8IAAC4GdBHAZDTcOYTAFzB+vXrVatWLRUsWPCKy4waNcr996uvvqqlS5eqZ8+eqlKlirZt26Y33nhD8fHx6tOnj3u5NWvWqHTp0ho6dKgsy9KoUaP09NNP6/PPP1dgYKB++uknde3aVbVq1dKECRN05MgRDRw40GO9lmWpT58+2rlzp/r166fSpUtr3bp1GjBggFJSUjw6nHPmzFGXLl305JNPKjQ01HtvEAAA8An6KAByGsInAMjEuXPndO7cOZUsWTLDvMsH8LTZbPrtt9+0cOFCDRw4UD169JAk3XfffbLZbJo6daoef/xxhYWFuV8/Y8YM5c2bV5KUkJCg5557Trt371aFChU0depUFSxYUFOmTFFQUJAkKSwsTAMGDHCv89tvv9XXX3+tsWPHuk+nj46OVlJSkmJjY/XQQw+5j4TefvvtGjRokHffIAAA4BP0UQDkRFx2BwCZcLlcmU4/dOiQ7rnnHo9HgwYNtHnzZlmWpZiYGDmdTvcjJiZGycnJ2rFjh7uMMmXKuDt1khQeHi5JSkpKkiTt2LFD0dHR7k6dJDVs2FCBgYHu55s2bZLNZtMDDzyQYX1//PGHfv75Z/ey5cqV886bAgAAfI4+CoCciDOfACATYWFhCgkJ0dGjRz2mFy1aVIsXL3Y/f+edd7Rv3z6dPXtWktS0adNMyzt+/Lj77zx58njMCwi4dBwgrTN57tw59xHINHa73WPa2bNnZVmWqlatmun6Tpw44e7QhYSEXHE7AQBAzkIfBUBORPgEAFcQExOjL774QufPn3cfBQwODlbFihXdy9x6662SpPz580uSPvjgg0zHLLj99tuzvN5bb71VJ0+e9JhmWZbOnTvnfp4vXz6FhIToww8/zLSMO++8M8vrAwAAOQt9FAA5DZfdAcAV9OjRQ06nU0OHDlVKSkqG+RcuXNDhw4clSdWrV5cknTlzRhUrVnQ/Tp8+rfHjx7uPOmZF7dq19dVXX7lPcZekr7/+WhcvXnQ/r1GjhhITE2VZlsf69u3bp3feeSfDmA8AAMB/0EcBkNNw5hMAXEFERITeeustDRkyRI888ohat26tiIgIOZ1Offfdd1q8eLFOnjyp7t27KyIiQs2bN9dLL72ko0ePqkKFCjpw4IDGjh2rYsWKZToo6JX06dNHn332mbp166bu3bvr9OnTGjdunMf4Cg888ICioqL01FNP6amnnlLp0qX1448/asKECYqOjlaBAgUMvCMAAOBmQB8FQE5D+AQAV9GoUSNVqFBB8+bN0+LFi3X06FFZlqXixYvrwQcfVLt27dydtjfffFNTp07V/PnzFRcXp4IFC+rBBx/Uv/71L4+BOK+lZMmSmj17tkaOHKkBAwaoYMGCeu655zRy5Ej3MgEBAZo2bZrGjx+vqVOn6tSpUwoPD1eXLl08bpkMAAD8E30UADmJzbIsy9eVAAAAAAAAgH9izCcAAAAAAAAYQ/gEAAAAAAAAYwifAAAAAAAAYAzhEwAAAAAAAIwhfAIAAAAAAIAxhE8AAAAAAAAwhvAJAAAAAAAAxhA+AQAAAAAAwBjCJwAAAAAAABhD+AQAAAAAAABjCJ8AAAAAAABgDOETAAAAAAAAjPl/sOwCwBOXIfsAAAAASUVORK5CYII=",
```

Comment:
I wonder if there's a way the embedded graphics could be SVGs? Not a high priority!